open-metadata-implementation/access-services/analytics-modeling/analytics-modeling-server/src/main/java/org/odpi/openmetadata/accessservices/analyticsmodeling/synchronization/model/AnalyticsAsset.java

```diff
@@ -150,5 +176,86 @@ public boolean hasMetadataModule() {
 				|| (item != null && !item.isEmpty());
 	}
 
+	@Override
+	public boolean equals(Object obj) {
+		
+		if (this == obj) {
+			return true;
+		}
+		
+		if (!(obj instanceof AnalyticsAsset)) {
+			return false;
+		}
+		
+		AnalyticsAsset asset = (AnalyticsAsset)obj;
+		
+		return Objects.equals(uid, asset.uid)
+				&& Objects.equals(location, asset.location)
+				&& Objects.equals(type, asset.type)
+				&& Objects.equals(lastModified, asset.lastModified)
+				&& super.equals(asset);
+	}
+	
+    /**
+     * Return a number that represents the contents of this object.
+     *
+     * @return int
+     */
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(super.hashCode(), uid, location, type, lastModified);
+    }
+	
+	/**
+	 * Check if internal data of the asset is modified.
+	 * @param asset to check.
+	 * @return true if modified.
+	 * @throws AnalyticsModelingCheckedException if different assets are compared. 
+	 */
+	public boolean isModified(AnalyticsAsset asset) throws AnalyticsModelingCheckedException {
```

Comment:
Should not have logic methods in Jackson beans since they can be called at odd instances.
This applied to all the non-getter/setter methods in the bean

Reply:
Logic not related to the bean but useful to work with the class is moved to utility class.

Reply:
It should stil lbe in another class.